tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,340 @@
+            break
+
+    if program_pid != -1:
+        logger.info("Found program '{}' in the '{}' state with pid {}"
+                    .format(program_name, program_status, program_pid))
+
+    return program_status, program_pid
+
+
+def get_container_autorestart_states(duthost):
+    """
+    @summary: Get container names and their autorestart states by analyzing
+              the command output of "show container feature autorestart"
+    @return:  A dictionary where keys are the names of containers which have the
+              autorestart feature implemented and values are the autorestart feature
+              state for that container
+    """
+    container_autorestart_states = {}
+
+    show_cmd_output = duthost.shell(CMD_CONTAINER_FEATURE_AUTORESTART)
+    for line in show_cmd_output["stdout_lines"]:
+        container_name = line.split()[0].strip()
+        container_state = line.split()[1].strip()
+        if container_state in ["enabled", "disabled"]:
+            container_autorestart_states[container_name] = container_state
+
+    return container_autorestart_states
+
+
+def get_disabled_container_list(duthost):
+    """
+    @summary: Get the container/service names which are disabled
+    @return: A list includes the names of disabled containers/services
+    """
+    disabled_containers = []
+
+    show_cmd_output = duthost.shell(CMD_FEATURE)
+    for line in show_cmd_output["stdout_lines"]:
+        container_name = line.split()[0].strip()
+        container_state = line.split()[1].strip()
+        if container_state == "disabled":
+            disabled_containers.append(container_name)
+
+    return disabled_containers
+
+
+def is_container_running(duthost, container_name):
+    """
+    @summary: Decide whether the container is running or not
+    @return:  Boolean value. True represents the container is running
+    """
+    result = duthost.shell("docker inspect -f \{{\{{.State.Running\}}\}} {}".format(container_name))
+    return result["stdout_lines"][0].strip() == "true"
+
+
+def check_container_state(duthost, container_name, should_be_running):
+    """
+    @summary: Determine whether a container is in the expected state (running/not running)
+    """
+    is_running = is_container_running(duthost, container_name)
+    return is_running == should_be_running
+
+
+def get_program_status(duthost, container_name, program_name):
+    """
+    @summary: Return the status of a program in the specified container
+    @return: "RUNNING" or "EXITED" represents the program is in the running
+             or exited status
+    """
+    program_status = None
+
+    process_list = duthost.shell("docker exec {} supervisorctl status".format(container_name))
+    for process_info in process_list["stdout_lines"]:
+        if process_info.find(program_name) != -1:
+            program_status = process_info.split()[1].strip()
+            break
+
+    return program_status
```

Comment:
It might be simpler to just use the function in process_util.py. Checking all critical processes might be more time consuming but it would be more complete. What if killing one thread caused cascade failures?

Reply:
I just realized this function isn't even used any longer. It should be removed.

Reply:
Good catch! Removed.